src/viking.js

```diff
@@ -1,11 +1,87 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength) {
+        this.health = health;
+        this.strength = strength;
+    }
+    attack() {
+        return this.strength;
+    }
+    receiveDamage(damage) {
+        this.health -= damage;
+    }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor(name, health, strength) {
+        super(health, strength);
+        this.name = name;
+    }
+
+    receiveDamage(damage) {
+        this.health -= damage;
+        if (this.health > 0) {
+            return `${this.name} has received ${damage} points of damage`
+        } else return `${this.name} has died in act of combat`;
+    }
+
+    battleCry() {
+        return "Odin Owns You All!";
+    }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+    receiveDamage(damage) {
+        this.health -= damage;
+        if (this.health > 0) {
+            return `A Saxon has received ${damage} points of damage`
+        } else return `A Saxon has died in combat`;
+    }
+}
+
 
 // War
-class War {}
+class War {
+    vikingArmy = [];
+    saxonArmy = [];
+    addViking(viking) {
+        this.vikingArmy.push(viking);
+    };
+    addSaxon(saxon) {
+        this.saxonArmy.push(saxon);
+    };
+    vikingAttack(){
+        const i = Math.floor(Math.random() * this.saxonArmy.length);
+        const randomSaxon = this.saxonArmy[i];
+        const j = Math.floor(Math.random() * this.vikingArmy.length);
```

Comment:
you use this way of creating random indeces 4 times, time to refactor it into it's own function ^^